### DISCUSSION TITLE
How do I add custom CSS at the end of the <head> ?

### DISCUSSION THREAD

**Discussion by srikanth912** [vercel/next.js#13533] [Help]
Hello All, 

We are building a multitenant application and would like to allow customers to overwrite the default theme CSS. For this we allow customers to add their own CSS which is retrieved via API as a variable (let's say `customCSS`).

We would like to insert this at the end of the `<head>` tag like `<style>{customCss}</>` after next.js generated styles (which is the default theme CSS). 

Tried below but it's not coming at the end (Stripped irrelevant meta tags).

```
import React from 'react';
import NextHead from 'next/head';

const SiteHead = (props) => {
  return (
    <NextHead>
      <style>
        {customCSS}
      </style>
    </NextHead>
  );
};
```

**Comment by srikanth912** [comment]
I've tried adding custom CSS in _document.js as well but the same results (custom CSS coming before other styles which are imported in _app.js). 

```
import React from 'react';
import Document, { Html, Head, Main, NextScript } from 'next/document';

const customCss = "body {color: red}"

class CustomDocument extends Document {
  static async getInitialProps(ctx) {
    const { html, head, styles } = await Document.getInitialProps(ctx);
    return { html, head, styles };
  }

  render() {
    return (
      <Html>
        <Head>
          <style>{customCss} </style>
        </Head>
        
        <body>
          <Main />
          <NextScript />
        </body>
      </Html>
    )
  }
}

```

**Comment by srikanth912** [comment]
Still stuck with this issue. Any help would be much appreciated.

**Comment by srikanth912** [comment]
Just bumping this thread so someone can point me in the right direction.

**Comment by jamesmosier** [comment]
As far as I know there is not way to specify the order an element gets added to `next/head`. Maybe someone else has ideas, but I would think you might be on your own as far as appending styles. Something like this:

```js
const SiteHead = (props) => {
  React.useEffect(() => {
    const style = document.createElement('style');
    style.innerText = 'body { color: green; }'
    const head = document.getElementsByTagName('head')[0];
    head.appendChild(style);
  }, [])

  return ( ...
```

**Comment by revskill10** [comment]
You can try adding this to `<Head>` component
```
<style
      id="holderStyle"
      dangerouslySetInnerHTML={{
         __html: `
         *, *::before, *::after {
           transition: none!important;
         }
         `,
      }}/>
```

**Reply by srikanth912** [reply]
I've tried this before and doesn't make any difference in the ordering.

**Comment by ArthurClemens** [comment]
If anyone arrives here from Google: use `style jsx` or `style jsx global`:

```jsx
<style jsx global>{`
  body {
    background: #000;
  }
`}</style>
```

**Reply by coderdix** [reply]
> If anyone arrives here from Google: use `style jsx` or `style jsx global`:
> 
> ```js
> <style jsx global>{`
>   body {
>     background: #000;
>   }
> `}</style>
> ```

i dont think this works with SSR sites :-(

**Comment by fmal** [comment]
@srikanth912 i think the solution in the following example would solve your case: https://github.com/ben-rogerson/twin.examples/blob/b52ac511ebf221471a01fea1c77d90b19a6eb5dc/next-stitches-typescript/pages/_document.tsx

**Reply by AndrewPrifer** [reply]
This.

**Reply by ajayjaggi97** [reply]
Thanks for this ❤️

**Comment by AdrianMachado** [comment]
In case people are looking for an Next JS 13 App Directory solution, I ended up having to use a [Script tag](https://nextjs.org/docs/app/api-reference/components/script#strategy) and wrote an inline script to add link tag that applied my CSS on hydration

**Reply by shamith09** [reply]
This is creating FOUC so if there's another solution which can do this at build time would greatly appreciate

**Reply by AdrianMachado** [reply]
I wish :( I do observe the FOUC but its not too bad once the site is cached. Alas, no better solution yet

**Reply by Enkratia** [reply]
Why would this cause FOUC if "beforeInteractive" strategy is used?

**Comment by petersvegrup** [comment]
I'm quite late to the party and this might already have been solved, but try using CSS cascade layers and add customCSS to a layer that is defined as last in the specificity order. https://developer.mozilla.org/en-US/docs/Web/CSS/@layer. Let me know if it works.

**Comment by marlonmarcello** [comment]
I am on the same boat as @petersvegrup . NextJs insists in loading the styles in the wrong order.
